File: festival-web/src/main/java/cn/bdqfork/web/util/XmlUtils.java
package cn.bdqfork.web.util;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.dataformat.xml.XmlMapper;

/**
 * @author bdq
 * @since 2020/2/20
 */
public class XmlUtils {
    private static XmlMapper xmlMapper = new XmlMapper();

    public static String toXml(Object instance) throws JsonProcessingException {
        return xmlMapper.writeValueAsString(instance);
    }

    public static <T> T fromXml(String xml, Class<T> clazz) throws JsonProcessingException {
        return xmlMapper.readValue(xml, clazz);
    }
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/util/XmlUtils.kt
package cn.bdqfork.kotlin.web.util

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.dataformat.xml.XmlMapper

/**
 * @author bdq
 * @since 2020/2/20
 */
object XmlUtils {
    private val xmlMapper = XmlMapper()
    @Throws(JsonProcessingException::class)
    fun toXml(instance: Any): String {
        return xmlMapper.writeValueAsString(instance)
    }

    @Throws(JsonProcessingException::class)
    fun <T> fromXml(xml: String, clazz: Class<T>): T {
        return xmlMapper.readValue(xml, clazz)
    }
}